server/plugins/client/index.js

```diff
@@ -31,28 +31,24 @@ function register (server, options, next) {
               return reject(error)
             }
 
-            resolve(bundleBuffer)
-
             if (!hasUpdate) {
               log.info('client', 'bundle is up to date')
-              return
-            }
-
-            if (options.config.inMemory) {
+            } else if (options.config.inMemory) {
               log.silly('client', 'running in memory, not writing bundle to ' + bundleTargetPath)
-              return
+            } else {
+              log.info('client', 'bundle is out of date')
+              log.silly('client', 'writing bundle to ' + bundleTargetPath)
+              fs.writeFile(bundleTargetPath, bundleBuffer, function (error) {
+                if (error) {
+                  log.warn('client', 'could not write to ' + bundleTargetPath + '. Bundle cannot be cached and will be re-generated on server restart')
+                  return
+                }
+
+                log.info('client', 'bundle written to ' + bundleTargetPath)
+              })
```

Comment:
@gr2m : I used another approach by creating an actual `decideAction`-Method to call. Now the code looks like before - only inside an own method. What do you think?